halpybot/packages/utils/decorators.py

```diff
@@ -22,166 +22,152 @@
-                    return await ctx.reply(
-                        get_help_text(ctx.bot.commandsfile, ctx.command)
-                    )
-                xcoord = args[0].strip()
-                ycoord = args[1].strip()
-                zcoord = args[2].strip()
-                try:
-                    float(xcoord)
-                    float(ycoord)
-                    float(zcoord)
-                except ValueError:
-                    return await ctx.reply("All coordinates must be numeric.")
-                try:
-                    return await function(ctx, xcoord, ycoord, zcoord)
-                except NoResultsEDSM:
-                    return await ctx.reply(
-                        "No system was found in the EDSM database for the given coordinates."
-                    )
-                except EDSMReturnError:
-                    logger.exception("Received a malformed reply from EDSM.")
-                    return await ctx.reply(
-                        "Received a reply from EDSM about the given coordinates, "
-                        "but could not process the return."
-                    )
-                except EDSMLookupError:
-                    logger.exception("Failed to query EDSM for coordinate details.")
-                    return await ctx.reply(
-                        "Failed to query EDSM for coordinate details."
-                    )
-
-            return guarded
-
-        return decorator
-
-    @staticmethod
-    def dist_exceptions():
-        """Handle the various possible EDSM Exceptions for Distance calculations"""
-
-        def decorator(function):
-            @functools.wraps(function)
-            async def guarded(ctx, args: List[str]):
-                if len(args) <= 1:  # Minimum Number of Args is 2.
+def sys_exceptions():
+    """Handle the various possible EDSM Exceptions for system calculations"""
+
+    def decorator(function):
+        @functools.wraps(function)
+        async def guarded(ctx, args: List[str]):
+            system, cache_override = await cache_prep(ctx, args)
+            cleaned_sys = await sys_cleaner(system)
+            try:
+                return await function(ctx, cleaned_sys, cache_override)
+            except NoResultsEDSM:
+                return await ctx.reply(
+                    f"No system and/or commander named {system} was found in the EDSM database."
+                )
+            except EDSMReturnError:
+                logger.exception("Received a malformed reply from EDSM.")
+                return await ctx.reply(
+                    f"Received a reply from EDSM about {system}, but could not process the return."
+                )
+            except EDSMLookupError:
+                logger.exception("Failed to query EDSM for coordinate details.")
+                return await ctx.reply("Failed to query EDSM for coordinate details.")
+
+        return guarded
+
+    return decorator
```

Comment:
This is a wrapping decorator that accepts no arguments, yet is written as if it accepted arguments.
this decorator can be simplified to simply:tm:  be the `def decorator` body and callsites simplified

callsites would be simplified to 
```py
@sys_exceptions
async def ....
```